### DISCUSSION TITLE
Request loops (e.g. for polling)

### DISCUSSION THREAD

**Discussion by riesinger** [Orange-OpenSource/hurl#1720] [Ideas] [answered]
Hi, thanks for this great tool!

My colleagues and I have the following idea/feature request I'd like to brainstorm.
We have APIs that reflect the status of "long" running processes, so the way you interact with them is through HTTP polling. As soon as the process is finished, the endpoint returns a changed body.

It would be awesome to test this API with hurl, but as we don't know how many polls it'll need to complete, we'd need some sort of looping mechanism, like the common `while` loop.

What do you think about this idea? I would be open at trying my hand at a bit of Rust and implementing something, but I'd love to hear your opinion on this first.

**Comment by jcamiel** [comment, marked as answer]
Hi @riesinger could you use the [`--retry`](https://hurl.dev/docs/manual.html#retry) option (either with the command line or with an option section)?

For instance:

```hurl
# Create a new job
POST http://api.example.org/jobs

HTTP 201
[Captures]
job_id: jsonpath "$.id"
[Asserts]
jsonpath "$.state" == "RUNNING"


# Pull job status until it is completed
GET http://api.example.org/jobs/{{job_id}}
[Options]
retry: 10   # maximum number of retry, -1 for unlimited

HTTP 200
[Asserts]
jsonpath "$.state" == "COMPLETED"
```

Basically, `--retry` will retry a request until all asserts (explicit and implicit like status code) are OK. So, you can make a condition on the response body and indefinitely loop until this condition is met. See the [documentation on retry](https://hurl.dev/docs/entry.html#retry)

Does it address your problematic?

**Reply by riesinger** [reply]
Yes, I didn't catch that in the docs, I'm sorry 😅 That's the exact solution I was looking for 🙂
